src/code/z_actor.c

```diff
@@ -471,237 +484,241 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
+                if (alpha < 0) {
+                    alpha = 0;
+                }
             }
         }
+    }
 
-        actor = targetCtx->unk_94;
-        if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_CANT_LOCK_ON)) {
-            TatlColor* color = &sTatlColorList[actor->category];
-
-            POLY_XLU_DISP = Gfx_SetupDL(POLY_XLU_DISP, SETUPDL_7);
+    actor = targetCtx->arrowPointedActor;
+    if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_CANT_LOCK_ON)) {
+        TatlColor* color = &sTatlColorList[actor->category];
 
-            Matrix_Translate(actor->focus.pos.x,
-                             actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y) + 17.0f,
-                             actor->focus.pos.z, MTXMODE_NEW);
-            Matrix_RotateYS((play->gameplayFrames * 3000), MTXMODE_APPLY);
-            Matrix_Scale((iREG(27) + 35) / 1000.0f, (iREG(28) + 60) / 1000.0f, (iREG(29) + 50) / 1000.0f,
-                         MTXMODE_APPLY);
+        POLY_XLU_DISP = Gfx_SetupDL(POLY_XLU_DISP, SETUPDL_7);
 
-            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, color->inner.r, color->inner.g, color->inner.b, 255);
-            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
-            gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
-        }
+        Matrix_Translate(actor->focus.pos.x, actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y) + 17.0f,
+                         actor->focus.pos.z, MTXMODE_NEW);
+        Matrix_RotateYS(play->gameplayFrames * 0xBB8, MTXMODE_APPLY);
+        Matrix_Scale((iREG(27) + 35) / 1000.0f, (iREG(28) + 60) / 1000.0f, (iREG(29) + 50) / 1000.0f, MTXMODE_APPLY);
 
-        CLOSE_DISPS(play->state.gfxCtx);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, color->inner.r, color->inner.g, color->inner.b, 255);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
+        gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
     }
+
+    CLOSE_DISPS(play->state.gfxCtx);
 }
 
 // OoT: func_8002C7BC
-void func_800B5814(TargetContext* targetCtx, Player* player, Actor* actor, GameState* gameState) {
-    PlayState* play = (PlayState*)gameState;
-    Actor* sp68 = NULL;
+void Target_Update(TargetContext* targetCtx, Player* player, Actor* targetedActor, PlayState* play) {
+    s32 pad;
+    Actor* actor = NULL;
     s32 category;
     Vec3f projectedPos;
     f32 invW;
 
+    // If no actor is currently targeted, then try to find a targetable actor
     if ((player->targetedActor != NULL) && (player->unk_AE3[player->unk_ADE] == 2)) {
-        targetCtx->unk_94 = NULL;
+        targetCtx->arrowPointedActor = NULL;
     } else {
-        func_800BB8EC(gameState, &play->actorCtx, &sp68, &D_801ED920, player);
-        targetCtx->unk_94 = sp68;
+        Target_GetTargetActor(play, &play->actorCtx, &actor, &D_801ED920, player);
+        targetCtx->arrowPointedActor = actor;
     }
```

Comment:
The comment at the top isn't quite right. `(player->unk_AE3[player->unk_ADE] == 2)` is specifically checking that you're holding down on the analog stick, so it will only not search for a target Actor if you're pressing down while locked on to an actor